scripts/rogue-keeper-upper.php

```diff
@@ -0,0 +1,300 @@
+<?php
+/**
+ * Keeper-Upper Script to export Signups, Reportbacks, and whatever else from drupal into Rogue.
+ * This is to be run on a recurring basis after the big migration has completed. 
+ *
+ * to run:
+ * drush --script-path=../scripts/ php-script rogue-keeper-upper.php
+ */
+
+// 1. New signups
+// Pick up from where we left off, if we want to
+$last_saved = variable_get('dosomething_rogue_last_signup_kept_up', NULL);
+
+if ($last_saved) {
+  $signups = db_query("SELECT signups.sid, signups.uid, signups.nid, signups.run_nid, signups.source, signups.timestamp, rb.quantity, rb.why_participated, rb.rbid, rb.flagged, rb.updated
+                      FROM dosomething_signup signups
+                      LEFT JOIN dosomething_reportback rb ON signups.uid = rb.uid AND signups.nid = rb.nid AND signups.run_nid = rb.run_nid
+                      WHERE signups.sid > $last_saved AND signups.sid NOT IN (SELECT sid FROM dosomething_rogue_signups)
+                      ORDER BY signups.sid");
+}
+else {
+  $signups = db_query('SELECT signups.sid, signups.uid, signups.nid, signups.run_nid, signups.source, signups.timestamp, rb.quantity, rb.why_participated, rb.rbid, rb.flagged, rb.updated
+                      FROM dosomething_signup signups
+                      LEFT JOIN dosomething_reportback rb ON signups.uid = rb.uid AND signups.nid = rb.nid AND signups.run_nid = rb.run_nid
+                      WHERE signups.sid NOT IN (SELECT sid FROM dosomething_rogue_signups)
+                      ORDER BY signups.sid');
+}
+
+$client = dosomething_rogue_client();
+
+foreach ($signups as $signup) {
+  $data = [];
+
+  echo 'Trying sid ' . $signup->sid . '...' . PHP_EOL;
+
+  $northstar_user = dosomething_northstar_get_user($signup->uid, 'drupal_id');
+
+  // Only try to send to Rogue if we have a Northstar ID
+  if (isset($northstar_user)) {
+    // Match Rogue's timestamp format
+    $created_at = date('Y-m-d H:i:s', $signup->timestamp);
+
+    // Format signup data
+    $data = [
+      'northstar_id' => $northstar_user->id,
+      'campaign_id' => $signup->nid,
+      'campaign_run_id' => $signup->run_nid,
+      'do_not_forward' => TRUE,
+      'source' => $signup->source,
+      'created_at' => $created_at,
+      'updated_at' => $created_at, // Set updated same as created, will be overwritten if there was a RB submitted
+    ];
+
+    // Include all the Reportback data and files if they exist
+    if ($signup->rbid) {
+      // Pull the quantity and why
+      $data['quantity'] = $signup->quantity;
+      $data['why_participated'] = $signup->why_participated;
+
+      // Match Rogue's timestamp format
+      $updated_at = date('Y-m-d H:i:s', $signup->updated);
+      $data['updated_at'] = $updated_at;
+    }
+
+    // Send the request to Rogue
+    try {
+      $response = $client->postSignup($data);
+
+      // Make sure we get a successful response
+      if ($response) {
```

Comment:
@DFurnes ahh and put all the rest after the try and catches? I think that would make it hard to check for a `404` because I'm not sure how to check if `$response` is truly not set because an error happened or if it's `null` because a `404` was returned.